subsys/net/lib/nrf_cloud/src/nrf_cloud_alerts.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <net/nrf_cloud.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <date_time.h>
+#include "nrf_cloud_fsm.h"
+#include "nrf_cloud_mem.h"
+#include "nrf_cloud_codec.h"
+#include "nrf_cloud_transport.h"
+#include <net/nrf_cloud_alerts.h>
+
+LOG_MODULE_REGISTER(nrf_cloud_alerts, CONFIG_NRF_CLOUD_LOG_LEVEL);
+
+static atomic_t alert_sequence;
+
+int nrf_cloud_alert_send(enum nrf_cloud_alert_type type, float value, const char *description)
+{
+	struct nct_dc_data output = {.message_id = NCT_MSG_ID_USE_NEXT_INCREMENT};
+	struct nrf_cloud_alert_info alert = {.ts_ms = 0, .type = type, .value = value,
+					     .description = description};
+	int64_t timestamp;
+	int err = 0;
+
+	if (nfsm_get_current_state() != STATE_DC_CONNECTED) {
+		LOG_ERR("Cannot send alert; not connected");
+		return -EACCES;
+	}
+
+	if (IS_ENABLED(CONFIG_DATE_TIME)) {
+		/* Acquire timestamp; if not available, send sequence */
+		err = date_time_now(&timestamp);
+		if (!err) {
+			alert.ts_ms = timestamp;
+		} else if ((err == -ENODATA) &&
+			 IS_ENABLED(CONFIG_NRF_CLOUD_ALERTS_SEQ_IF_NO_TIME)) {
+			/* time not available yet; use sequence instead */
+			alert.sequence = alert_sequence++;
+		} else {
+			LOG_ERR("Error requesting current time: %d", err);
+			return err;
+		}
+	} else {
+		alert.sequence = alert_sequence++;
```

Comment:
re: comment about `CONFIG_NRF_CLOUD_ALERTS_SEQ_IF_NO_TIME`

how about just checking if `ts_ms` is zero and setting the sequence number in that case.

the kconfig option could just control the return value at the end of the function. 
if the user is very concerned about the timestamp, the function would return `ETIME` if no timestamp was used (but the alert would still be sent).

Reply:
I think I'll just scrap the `CONFIG_NRF_CLOUD_ALERTS_SEQ_IF_NO_TIME` and use `seq` if `ts_ms` is zero.

Reply:
Would it make any sense to ***_always_*** set the sequence number here, and have the codec be responsible for deciding whether to append `ts`, `seq`, or both, depending on KConfig args and whether the timestamp is, as you say, zero?

Basically, reinterpret `CONFIG_NRF_CLOUD_ALERTS_SEQ_XXX` as format specifiers rather than counting behavior specifiers

Reply:
I didn't see your above comment while I was reviewing, sorry.

I don't think it's a bad idea to allow this behavior to be configurable

Reply:
I rethought this after Justin's comments, @glarsennordic .  I'd rather alerts be more guaranteed to get sent, whether, for example, the NTP server is responding or not.  People shouldn't use alerts unless there is a value to transmitting them to the cloud, so we should try hard to make sure they can get sent.

Reply:
Oh I missed that else clause that could cause the alert to completely fail to send. Yes, I agree, they should always be sent. I'm just talking more specifically about whether and when sequence numbers should be added / incremented

Reply:
I've changed it to have a different CONFIG, `CONFIG_NRF_CLOUD_ALERTS_SEQ_ALWAYS`.  If enabled, there will always be a seq number added; otherwise, only if there is no timestamp.  This would be useful when the absolute order that alerts are issued is really important to a customer.